app/src/org/commcare/util/externalizable/AndroidClassHasher.java

```diff
@@ -13,14 +13,13 @@
 
     private static final int CLASS_HASH_SIZE = 4;
 
-    MessageDigest mMessageDigester;
+    private final MessageDigest mMessageDigester;
     
     public AndroidClassHasher() {
         try {
             mMessageDigester = java.security.MessageDigest.getInstance("MD5");
         } catch (NoSuchAlgorithmException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
+            throw new RuntimeException(e);
```

Comment:
In order to synchronize safely over `mMessageDigester` I made it `final`. That meant that I needed to somehow set `mMessageDigester` or abort when the `NoSuchAlgorithmException` is raised. I chose to abort, since the whole system probably won't work if we don't have a class hasher. 

Do you agree with my decision?